tapir/shifts/templatetags/shifts.py

```diff
@@ -119,25 +120,21 @@ def get_html_classes_for_filtering(shift: Shift) -> set:
         return {"is_in_the_past"}
 
     filter_classes = set()
-    num_valid_attendances = 0
     for slot in shift.slots.all():
         valid_attendance = None
         for a in slot.attendances.all():
             if a.is_valid():
                 valid_attendance = a
                 break
 
-        if valid_attendance and valid_attendance.state == ShiftAttendance.State.PENDING:
-            num_valid_attendances += 1
-
         if (
             not valid_attendance
             or valid_attendance.state == ShiftAttendance.State.LOOKING_FOR_STAND_IN
         ):
             filter_classes.add("freeslot_any")
             filter_classes.add("freeslot_" + shift_name_as_class(slot.name))
 
-    if num_valid_attendances < shift.get_num_required_attendances():
```

Comment:
While it does look nicer with `is_understaffed()`, that leads to many more database requests and render the calendar view unusable on the prod server. It's a bit hidden sorry!

You can check that by setting `ENABLE_SILK_PROFILING` to `True` in the settings file, then applying migrations, then checking http://localhost:8000/silk/requests/ . On master, I get 18 db queries when visiting the calendar page, on this branch I get 163.

Reply:
Oh I didn't know, thanks for pointing out